torch/distributions/relaxed_categorical.py

```diff
@@ -0,0 +1,110 @@
+import torch
+from torch.autograd import Variable
+from torch.nn.functional import log_softmax
+from torch.distributions import constraints
+from torch.distributions.utils import clamp_probs, broadcast_all
+from torch.distributions.categorical import Categorical
+from torch.distributions.distribution import Distribution
+from torch.distributions.transformed_distribution import TransformedDistribution
+from torch.distributions.transforms import ExpTransform
+
+
+def _log_softmax(x, *args, **kwargs):
+    if isinstance(x, torch.Tensor):
+        return log_softmax(Variable(x), *args, **kwargs).data
+    return log_softmax(x, *args, **kwargs)
+
+
+class ExpRelaxedOneHotCategorical(Distribution):
```

Comment:
`RelaxedCategorical` seems ok too, since `OneHot` is kind of the only relay to relax.